paddle/cuda/src/hl_cuda_cnn.cu

```diff
@@ -145,24 +145,28 @@ void hl_shrink_col2feature(const real * dataCol, size_t channels,
   CHECK_SYNC("hl_shrink_col2feature failed");
 }
 
-__global__ void KeMaxPoolForward(int nthreads, const real* inputData,
-                                 int channels, int height, int width,
-                                 int pooledH, int pooledW,
-                                 int ksize, int stride, int start,
+__global__ void KeMaxPoolForward(const int nthreads, const real* inputData,
+                                 const int channels, const int height,
+                                 const int width,
+                                 const int pooledH, const int pooledW,
+                                 const int ksizeW, const int ksizeH,
+                                 const int strideH, const int strideW,
+                                 const int offsetH, const int offsetW,
                                  real* tgtData) {
-  int index =  blockIdx.y * blockDim.x + threadIdx.x;
+  int index =  blockIdx.x * blockDim.x + threadIdx.x;
   if (index < nthreads) {
     int pw = index % pooledW;
     int ph = (index / pooledW) % pooledH;
     int c = (index / pooledW / pooledH) % channels;
-    int frameNum = blockIdx.x;
-    int hstart = ph * stride + start;
-    int hend = min(hstart + ksize, height);
-    int wstart = pw * stride + start;
-    int wend = min(wstart + ksize, width);
+    int frameNum = index / pooledW / pooledH / channels;
```

Comment:
The original implementation each block only calculate one sample, it is easy to support leading dimension for image size.